### DISCUSSION TITLE
Error executing ListObjectsV2 storing backups on S3

### DISCUSSION THREAD

**Discussion by massiws** [spatie/laravel-backup#1588] [Q&A] [answered]
Installed laravel-backup v.7 into a Laravel 8 project with `league/flysystem-aws-s3-v3:"^1.0"`.
Backups works well on local storage, bat fails on S3.

This is the error:
`Copying zip failed because: Error executing "ListObjectsV2" on "//mybucket./?list-type=2&prefix=backup-folder%2F2022-10-06-10-25-15.zip%2F&max-keys=1"; AWS HTTP error: cURL error 6: Could not resolve host: mybucket.`

`ListObjectsV2` don't appear on AWS IAM policy operations, so I try to allow all S3 operations:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:*",
                "s3-object-lambda:*"
            ],
            "Resource": "*"
        }
    ]
}
```

but I get the same error.
Can be the `league/flysystem-aws-s3-v3` package used on in new version (2.x or 3.x) also with `laravel-backup` v.7?
Have I missing something else?

**Comment by erikn69** [comment]
> Can be the league/flysystem-aws-s3-v3 package used on in new version (2.x or 3.x) also with laravel-backup v.7?

Not, for flysystem 2.x or 3.x must use Laravel-Backup 8.x, also v.7 has no support at this time

**Reply by massiws** [reply]
@erikn69 , thanks for the quick response, but I'm confused...
This is `composer.json` for `laravel-backup` v.7:

```
    "require-dev": {
        // ...
        "league/flysystem-aws-s3-v3": "^1.0.29",
        // ...
    },
```
and this for  `laravel-backup` v.8:

```
    "require-dev": {
        // ...
        "league/flysystem-aws-s3-v3": "^2.0|^3.0",
        // ...
    },
```
I must use `laravel-backup` v.7 because my project is on Laravel 8: can I adopt `flysystem-aws-s3-v3` v2.0 (or 3.0) ?

**Reply by erikn69** [reply]
Read again my last answer 
> Not, for flysystem 2.x or 3.x you must use Laravel-Backup 8.x

https://github.com/thephpleague/flysystem-aws-s3-v3/commit/d75962eef01fbf774ce6d6b6855bcb7985de64ae try an older version of `flysystem-aws-s3-v3`

**Reply by massiws** [reply]
Sorry, I misunderstood.

> also v.7 has no support at this time

are you saying that I can't save backups on S3 with Laravel 8.x + laravel-backup:7.x + flysystem-aws-s3-v3: ^1.x ?

**Comment by massiws** [comment, marked as answer]
Finally solved: the problem was a misconfiguration of Spatie laravel-backup. 
Thanks.
